documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,296 @@
+ Removing this field means having no metrics exposed.
+ 
+The following is an example of a ConfigMap for a Kakfa cluster.
+
+.Example Kafka cluster ConfigMap
+[source,yaml,options="nowrap"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cluster
+  labels:
+    strimzi.io/kind: cluster
+    strimzi.io/type: kafka
+data:
+  kafka-nodes: "3"
+  kafka-image: "strimzi/kafka:latest"
+  kafka-healthcheck-delay: "15"
+  kafka-healthcheck-timeout: "5"
+  zookeeper-nodes: "1"
+  zookeeper-image: "strimzi/zookeeper:latest"
+  zookeeper-healthcheck-delay: "15"
+  zookeeper-healthcheck-timeout: "5"
+  KAFKA_DEFAULT_REPLICATION_FACTOR: "3"
+  KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: "3"
+  KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: "3"
+  kafka-storage: |-
+    { "type": "ephemeral" }
+  zookeeper-storage: |-
+    { "type": "ephemeral" }
+  kafka-metrics-config: |-
+    {
+      "lowercaseOutputName": true,
+      "rules": [
+          {
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*><>Count",
+            "name": "kafka_server_$1_$2_total"
+          },
+          {
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*, topic=(.+)><>Count",
+            "name": "kafka_server_$1_$2_total",
+            "labels":
+            {
+              "topic": "$3"
+            }
+          }
+      ]
+    }
+  zookeeper-metrics-config: |-
+    {
+      "lowercaseOutputName": true
+    }
+----
+
+==== Storage
+
+Both Kafka and Zookeeper save data to files.
+
+The Strimzi allows to save such a data in an "ephemeral" way (using `emptyDir`) or in a "persistent-claim" way using persistent
+volumes.
+It's possible to provide the storage configuration in the related ConfigMap using a JSON string as value for the 
+`kafka-storage` and `zookeeper-storage` fields.
+
+IMPORTANT: The `kafka-storage` and `zookeeper-storage` fields can't be changed when the cluster is up and running.
+
+The JSON representation has a mandatory `type` field for specifying the type of storage to use ("ephemeral" or "persistent-claim").
+
+The "ephemeral" storage is really simple to configure and the related JSON string is something like this.
+
+.Ephemeral storage JSON
+[source,json]
+----
+{ "type": "ephemeral" }
+
+----
+
+In case of "persistent-claim" type the following fields can be provided as well :
+
+* `size`: defines the size of the persistent volume claim (i.e 1Gi)
+* `class` : the Kubernetes/OpenShift https://kubernetes.io/docs/concepts/storage/storage-classes/[storage class] to use for dynamic volume allocation
+* `selector`: allows to select a specific persistent volume to use. It contains a `matchLabels` field which defines an
+inner JSON object with key:value representing labels for selecting such a volume.
+* `delete-claim`: specifies if the persistent volume claim has to be deleted when the cluster is un-deployed. Default is `false`
+
+Other than `type`, the `size` is the only mandatory one.
```

Comment:
Personally I prefer the mandatory/optional in the text for each config. It's easy to miss if it's in a different paragraph.